node/core/pvf/src/host.rs

```diff
@@ -433,6 +435,10 @@ async fn handle_precheck_pvf(
 	pvf: Pvf,
 	result_sender: PrepareResultSender,
 ) -> Result<(), Fatal> {
+	let pvf = match pvf {
+		Pvf::Code(code) => code,
+		Pvf::Hash(_) => todo!(),
+	};
```

Comment:
@pepyakin we could either expect `PvfCode` here or allow requesting by hash.

Reply:
Why not handle this outside of this fn and just expected `PvfCode` as input?

Reply:
We could possibly accept the hash here and look for it in artifacts cache, return `ArtifactNotFound` otherwise.
It's left as a question and thus `todo`

Reply:
I decided to keep it as is: request the code from the runtime right away:
https://github.com/paritytech/polkadot/blob/e4edd445f472a2610b25f1aaee525f9e16ed95fe/node/core/candidate-validation/src/lib.rs#L318-L321